pkg/grpc/requestinfo/requestinfo.go

```diff
@@ -76,6 +77,11 @@ func (s Source) GetSourceIP() string {
 	)
 }
 
+// HeaderGetter interface abstracts http.Header and metadata.MD types.
+type HeaderGetter interface {
+	Get(key string) []string
+}
```

Comment:
Move the interface and everything related to it (`GetFirst()`, type wrappers if any) to a separate file?